ethereum/core/src/test/java/org/hyperledger/besu/ethereum/bonsai/trielog/TrieLogLayerTests.java

```
@@ -0,0 +1,107 @@
+    Address address = Address.fromHexString("0x00");
+    StateTrieAccountValue oldValue = new StateTrieAccountValue(0, Wei.ZERO, Hash.EMPTY, Hash.EMPTY);
+    StateTrieAccountValue newValue =
+        new StateTrieAccountValue(1, Wei.fromEth(1), Hash.EMPTY, Hash.EMPTY);
+
+    Address otherAddress = Address.fromHexString("0x000000");
+    StateTrieAccountValue otherOldValue =
+        new StateTrieAccountValue(0, Wei.ZERO, Hash.EMPTY, Hash.EMPTY);
+    StateTrieAccountValue otherNewValue =
+        new StateTrieAccountValue(1, Wei.fromEth(1), Hash.EMPTY, Hash.EMPTY);
+
+    trieLogLayer.addAccountChange(address, oldValue, newValue);
+    otherTrieLogLayer.addAccountChange(otherAddress, otherOldValue, otherNewValue);
+
+    Assertions.assertThat(trieLogLayer).isEqualTo(otherTrieLogLayer);
+
+    Optional<StateTrieAccountValue> priorAccount = trieLogLayer.getPriorAccount(address);
+    Assertions.assertThat(priorAccount).isPresent();
+    Assertions.assertThat(priorAccount.get()).isEqualTo(oldValue);
+
+    Optional<StateTrieAccountValue> updatedAccount = trieLogLayer.getAccount(address);
+    Assertions.assertThat(updatedAccount).isPresent();
+    Assertions.assertThat(updatedAccount.get()).isEqualTo(newValue);
+  }
+
+  @Test
+  public void testAddCodeChange() {
+    Address address = Address.fromHexString("0xdeadbeef");
+    Bytes oldValue = Bytes.fromHexString("0x00");
+    Bytes newValue = Bytes.fromHexString("0x01030307");
+    Hash blockHash = Hash.fromHexStringLenient("0xfeedbeef02dead");
+
+    Address otherAddress = Address.fromHexString("0x0000deadbeef");
+    Bytes otherOldValue = Bytes.fromHexString("0x0000");
+    Bytes otherNewValue = Bytes.fromHexString("0x0001030307");
+    Hash otherBlockHash = Hash.fromHexStringLenient("0x00feedbeef02dead");
+
+    trieLogLayer.addCodeChange(address, oldValue, newValue, blockHash);
+    otherTrieLogLayer.addCodeChange(otherAddress, otherOldValue, otherNewValue, otherBlockHash);
```

so is this testing that the leading zeros don't make a difference?

---

I wanted an inexpensive way to ensure different objects are generated so we don't short-circuit the equals comparison.  the leading bytes don't matter for the address, but they do for the code bytes.  Fixed.

---

weren't the tests passing before with the extra leading zeros though?     Assertions.assertThat(trieLogLayer).isEqualTo(otherTrieLogLayer);

---

The failure was being masked in CI by spdx header failure for the new test class